lib/DragNDrop.pm

```diff
@@ -0,0 +1,205 @@
+################################################################################
+# WeBWorK Online Homework Delivery System
+# Copyright &copy; 2000-2021 The WeBWorK Project, http://openwebwork.sf.net/
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of either: (a) the GNU General Public License as published by the
+# Free Software Foundation; either version 2, or (at your option) any later
+# version, or (b) the "Artistic License" which comes with this package.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See either the GNU General Public License or the
+# Artistic License for more details.
+################################################################################
+
+=head1 NAME
+
+DragNDrop.pm - Drag-N-Drop Module
+
+=head1 DESCRIPTION
+
+DragNDrop.pm is a backend Perl module which facilitates the implementation of
+'Drag-And-Drop' in WeBWorK problems. It is meant to be used by other perl macros
+such as draggableProof.pl and draggableSubsets.pl
+
+=head1 TERMINOLOGY
+
+An HTML element into or out of which other elements may be dragged will be called a "bucket".
+An HTML element which houses a collection of buckets will be called a "bucket pool".
+
+=head1 USAGE
+
+Each macro aiming to implement drag-n-drop features must call at its initialization:
+ADD_CSS_FILE("https://cdnjs.cloudflare.com/ajax/libs/nestable2/1.6.0/jquery.nestable.min.css", 1);
+ADD_JS_FILE("https://cdnjs.cloudflare.com/ajax/libs/nestable2/1.6.0/jquery.nestable.min.js", 1);
+ADD_CSS_FILE("js/apps/DragNDrop/dragndrop.css", 0);
+ADD_JS_FILE("js/apps/DragNDrop/dragndrop.js", 0, { defer => undef });
+
+To initialize a bucket pool, do:
+
+my $bucket_pool = new DragNDrop($answerInputId, $aggregateList);
+
+$answerInputId is a unique identifier for the bucket_pool, it is recommended that
+it be generated with NEW_HIDDEN_ANS_NAME.
```

Comment:
Another minor issue.  There is no such subroutine NEW_HIDDEN_ANS_NAME.  This should be NEW_ANS_NAME.

Reply:
I have updated the files in the latest PR based on your recommendations, thanks!